lib/src/shared/codec_registry.dart

```diff
@@ -0,0 +1,37 @@
+import 'codec.dart';
+
+/// Encloses classes related to the compression and decompression of messages.
+class CodecRegistry {
+  CodecRegistry({List<Codec> codecs = const [const Identity()]})
+      : assert(codecs != null) {
```

Comment:
I'd write something like this instead:

```dart
CodecRegistry({List<Codec> codecs = ...})
    : assert(...),
      _codecs = Map.fromIterable(codecs, key: (c) => c.encodingName),
      supportedEncodings = codecs.map((c) => c.encodingName).toSet().join(',') {
  if (_codecs.length != codecs.length) {
    throw ArgumentError.value(codecs, 'codecs', 'contains multiple entries with the same name');
  }
  if (supportedEncodings.split(',').length != codecs.length) {
    throw ArgumentError.value(codecs, 'codecs', 'contains entries with names containing ","');
  } 
}

final Map<String, Codec> _codecs;
final String supportedEncodings;
```

Reply:
Changed as per suggestion but moved the second ArgumentError above in the map since split(',') on empty string returns a List of size 1.